cns/nmagent/client.go

```diff
@@ -84,11 +93,18 @@ func JoinNetwork(networkID, joinNetworkURL string) (*http.Response, error) {
 }
 
 // PublishNetworkContainer publishes given network container
-func PublishNetworkContainer(networkContainerID, createNetworkContainerURL string, requestBodyData []byte) (*http.Response, error) {
+func PublishNetworkContainer(networkContainerID, associatedInterfaceID, accessToken string, requestBodyData []byte) (*http.Response, error) {
 	logger.Printf("[NMAgentClient] PublishNetworkContainer NC: %s", networkContainerID)
 
+	createNCURL := fmt.Sprintf(
```

Comment:
I think the URLs should be built from a url.URL, not by string templating

Reply:
a common pattern for a client pkg is to have a baseURL that has the scheme and host set (and port if required), make copies of it in the client methods, then set the paths and queries on the copies and use url.String() to make the final input string for the http call.

Reply:
Changed it to incorporate some of this feedback